client/pd_service_discovery.go

```diff
@@ -991,30 +991,46 @@ func (c *pdServiceDiscovery) updateURLs(members []*pdpb.Member) {
 	log.Info("[pd] update member urls", zap.Strings("old-urls", oldURLs), zap.Strings("new-urls", urls))
 }
 
-func (c *pdServiceDiscovery) switchLeader(url string) (bool, error) {
+// switchLeader switches the leader of the PD cluster.
+// Note: For current implementation, when initializing the client, the connection to leader should be established.
+// Otherwise, the initialization will fail.
+func (c *pdServiceDiscovery) switchLeader(url string) (change bool, err error) {
 	oldLeader := c.getLeaderServiceClient()
 	if url == oldLeader.GetURL() && oldLeader.GetClientConn() != nil {
 		return false, nil
 	}
 
-	newConn, err := c.GetOrCreateGRPCConn(url)
+	var newConn *grpc.ClientConn
+	newConn, err = c.GetOrCreateGRPCConn(url)
 	// If gRPC connect is created successfully or leader is new, still saves.
-	if url != oldLeader.GetURL() || newConn != nil {
+	if url != oldLeader.GetURL() {
+		change = true
```

Comment:
This check and introducing `change` seem unnecessary due to line 999.

Reply:
If the client continues to fail to establish a connection with the leader, there will be situations where `oldURL` == `url` and `err != nil``. So it's not always `change = true` after line 999